homeassistant/components/bzutech/__init__.py

```diff
@@ -0,0 +1,82 @@
+"""The BZUTech integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+from bzutech import BzuTech
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import CONF_CHIPID, CONF_SENSORNAME, DOMAIN
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up BZUTech from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+    hass.data["host"] = "bzutech"
+    bzucoordinator = BzuCloudCoordinator(
+        hass=hass,
+        chipid=entry.data[CONF_CHIPID],
+        sensor=entry.data[CONF_SENSORNAME],
+        email=entry.data[CONF_EMAIL],
+        password=entry.data[CONF_PASSWORD],
+    )
+    hass.data[DOMAIN][entry.entry_id] = bzucoordinator
```

Comment:
please do a ` await bzucoordinator.async_config_entry_first_refresh()` before adding it to hass